### PR TITLE
fix: Bring back federated editing in viewer iframe

### PR DESCRIPTION
**Pull request by juliushaertl**
Quick attempt to fix https://github.com/nextcloud/richdocuments/issues/3605

Ideally we'd also remove the document.js usage there, but this at least brings back the functionality.

- [x] Try to come up with a cypress test

### CODE REVIEW

File: lib/Controller/DocumentController.php [hunk 1]
```diff
@@ -294,6 +296,16 @@ public function remote(string $shareToken, string $remoteServer, string $remoteS
 		return new TemplateResponse('core', '403', [], 'guest');
 	}
 
+	/**
+	 * Open file on Source instance with token from Initiator instance
+	 */
+	#[PublicPage]
+	#[NoCSRFRequired]
+	public function remotePost(string $shareToken, string $remoteServer, string $remoteServerToken, ?string $filePath = null): TemplateResponse {
+		return $this->remote($shareToken, $remoteServer, $remoteServerToken, $filePath);
```
Comment (juliushaertl) [inline lib/Controller/DocumentController.php:305]: This is needed as currently the routing cannot handle two routes with the same controller method name. We need one for POST and one for GET

File: src/helpers/coolParameters.js [hunk 2]
```diff
@@ -14,7 +14,7 @@ const getUIDefaults = () => {
 	const uiMode = defaults.UIMode ?? 'notebookbar'
 
 	const systemDarkMode = window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches
-	const dataset = (document.body.dataset.themes ? document.body.dataset : parent?.document.body.dataset) ?? {}
+	const dataset = {} // (document.body.dataset.themes ? document.body.dataset : parent?.document.body.dataset) ?? {}
```
Comment (juliushaertl) [inline src/helpers/coolParameters.js:17]: Needs to be checked still

**Review by mejo- — APPROVED** [review]
Didn't test and don't know the code well, but at least didn't spot anything obvious.

**Pull request merged**